src/librustc_mir/interpret/place.rs

```diff
@@ -354,7 +354,8 @@ where
         let (meta, offset) = if field_layout.is_unsized() {
             // re-use parent metadata to determine dynamic field layout
             let (_, align) = self.size_and_align_of(base.meta, field_layout)?
-                .expect("Fields cannot be extern types");
+                // If this is an extern type, we fall back to its static size and alignment.
+                .unwrap_or_else(|| base.layout.size_and_align());
```

Comment:
Interestingly, we can make `size_and_align_of` error, but have a separate `align_of` that works even for `extern type`. Only the alignment is used here, anyway!

Reply:
Oh, you can also just move the `Option` to be around the size, but not the alignment.

Reply:
I could, but I'd rather wait for the decision to treat the `size_of_val` and `align_of_val` intrinsics asymmetrically before doing that.

`extern type` will likely get alignment 1 currently, which is likely not actually correct for the type they are opaquely representing, so I am a bit worried about unilaterally implementing such behavior.

Reply:
You can't access that alignment *without* first having a reference, so having it too high would be a problem (the assumption might not be guaranteed by what you're FFI-ing to), but having it too low wouldn't break anything (other than field offsets).

So the way I see it, the alignment is only a lower bound, and `1` is fine, unless you want to use it as an aligned field in a struct.

Anyway, my concern is also kind of a confusing code thing.
You could also add a `.map(|(_, align)| align)` before `unwrap_or_else`, which would mean the code and the comment wouldn't mention the "size" of the `extern type`.

Reply:
> You can't access that alignment without first having a reference, so having it too high would be a problem (the assumption might not be guaranteed by what you're FFI-ing to), but having it too low wouldn't break anything (other than field offsets).

*Oh* that's a good point. We cannot create such references ourselves anyway as we cannot have data of that type.

Still I think miri shouldn't move ahead of the rest of the decision process here.

I will implement your `map` suggestion.

Reply:
I implemented what you suggested.